analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -81,18 +71,17 @@ static bool MethodDereferencesArguments(BaseMethodDeclarationSyntax method)
 
     protected override ProgramState PreProcessSimple(SymbolicContext context)
     {
-        var operation = context.Operation.Instance;
-        if (operation.Kind == OperationKindEx.ParameterReference
-            && operation.ToParameterReference().Parameter is var parameter
-            && !parameter.Type.IsValueType
-            && IsParameterDereferenced(context.Operation)
-            && NullableStateIsNotKnownForParameter(parameter)
-            && !parameter.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_FromServicesAttribute))
+        if (NullDereferenceCandidate(context.Operation.Instance) is { } candidate
+            && candidate.Kind == OperationKindEx.ParameterReference
+            && candidate.ToParameterReference() is { Parameter: var parameterSymbol, WrappedOperation: var parameterReference }
```

Comment:
The null check is not needed, the `var`s are abusing pattern matching in a way that we don't use in the code base. It makes it harder to read for me, because I need to track more variables that do not say anything new that would be useful
```suggestion
            && candidate.ToParameterReference() is var parameterReference
```